internal/controller/postgrescluster/postgres.go

```diff
@@ -375,6 +376,36 @@ func (r *Reconciler) reconcilePostgresUserSecrets(
 			))
 	}
 
+	// Sorts the slice of secrets.Items based on secrets with identical labels
+	// If one secret has "pguser" in its name and the other does not, the
+	// one without "pguser" is moved to the front.
+	// If both secrets have "pguser" in their names or neither has "pguser", they
+	// are sorted by creation timestamp.
+	// If two secrets have the same creation timestamp, they are further sorted by name.
+	// The secret to be used by PGO is put at the end of the sorted slice.
+	sort.Slice(secrets.Items, func(i, j int) bool {
+		// Check if either secrets have "pguser" in their names
+		isIPgUser := strings.Contains(secrets.Items[i].Name, "pguser")
+		isJPgUser := strings.Contains(secrets.Items[j].Name, "pguser")
```

Comment:
🌱 Seeing this reminded me that we only really care about picking the right secret _for each user_. That is, the order between different users doesn't matter. 🤔 That might simplify things here, but I'm not sure.